cleaner.py

```diff
@@ -0,0 +1,194 @@
+"""
+    Remove unwanted files from DFS or S3 by arguments :
+        - DFS:      Local certificate folders generated before uploading to S3
+        - AWS/S3:   .PNG files of Learner Certificate
+
+    Usage:
+        (certs) certs@learning-tribes:~$ pwd
+        /edx/app/certs
+        $ sudo -H -u certs bash
+        (certs) certs@learning-tribes:~$ source /edx/app/certs/venvs/certs/bin/activate
+        (certs) certs@learning-tribes:~$ /edx/app/certs/venvs/certs/bin/python /edx/app/certs/certificates/cleaner.py --target_type=[s3|dfs] --dryrun=[true|false]
+        [INFO] Dryrun Mode=True | cleaning AWS/S3 files, Bucket Name=lt-learning-customer2-default
+        Done !
+
+"""
+from abc import ABCMeta, abstractmethod
+from argparse import ArgumentParser
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+import os
+import shutil
+from sys import exit as process_terminate
+from traceback import format_exc
+import uuid
+
+import boto.s3
+from boto.s3.key import Key
+
+import settings
+
+
+class _CleanerInterface(object):
+    """Regulate all subclass of cleaner
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def run(self):
+        """Execute cleaning task
+        """
+        raise NotImplementedError
+
+
+class DFSCleaner(_CleanerInterface):
+    """Clean unwanted files in DFS ( folders: /downloads + /cert ).
+        And we can specify the `End Date` of date range in the process.
+    """
+    TARGET_ROOT_FOLDERS = [                 # Target folders where we want to clean
+        '/edx/var/certs/www-data/downloads',
+        '/edx/var/certs/www-data/cert'
+    ]
+    PERIOD_START_DATE = datetime(2020, 2, 15)
+    PERIOD_END_DATE = None
+
+    def __init__(self, dryrun=True):
+        self.PERIOD_END_DATE = datetime.now() - relativedelta(
```

Comment:
I would prefer to enter a _YYYY-MM-DD_ date like "2019-12-06" and we remove everything that is ≥ to this date. We don't need a range.
As for `dryrun`, it would be better to set the argument in the command line and then pass it to `cleaner.run()`.